File: bstree.py
class Node:
  
  def __init__( self, key ):
    self.key = key
    self.left = None
    self.right = None

  def insert( self, key ):

    if key == self.key:
      return

    if key > self.key:
      if self.right != None:
        self.right.insert( key )
      else:
        self.right = Node( key )
    else:
      if self.left != None:
        self.left.insert( key )
      else:
        self.left = Node( key )

  def print( self ):
    if self.left != None:
      self.left.print()
    
    print( self.key, end=' ' )

    if self.right != None:
      self.right.print()


class Tree:
  
  root = None
  
  def insert( self, key ):
    if self.root == None:
      self.root = Node( key )
    else:
      self.root.insert( key )
    
  def print( self ):
    if self.root != None:
      self.root.print()


tree = Tree()

for i in range(10):
  tree.insert( i )

tree.print()

File: bstree.js

var Node = function( key )
{
	this.key = key;
	this.left = null;
	this.right = null;
};

var Tree = function( )
{
	this.root = null;
};

Tree.prototype._insert = function( node, key )
{
	if( node == null )
	{
		return new Node( key );
	}

	if( node.key > key )
	{
		node.left = this._insert( node.left, key )
	}
	else if( node.key < key )
	{
		node.right = this._insert( node.right, key )
	}

	return node;
};

Tree.prototype.insert = function( key )
{
	this.root = this._insert( this.root, key );
};

Tree.prototype._print = function( node )
{
	if( node == null )
		return;
	this._print( node.left );
	console.debug( node.key );
	this._print( node.right );
};

Tree.prototype.print = function()
{
	this._print( this.root );
}

var tree = new Tree();
tree.insert( 5 );
tree.insert( 2 );
tree.insert( 1 );

console.debug( tree );